File: android/src/main/java/com/mycompany/plugins/example/PaysafePluginPlugin.java
package com.mycompany.plugins.example;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PaysafePlugin")
public class PaysafePluginPlugin extends Plugin {

    private final PaysafePlugin implementation = new PaysafePlugin();

    @Override
    public void load() {
        super.load();
        implementation.setupSDK();
    }

    @Override
    protected void handleOnStart() {
        super.handleOnStart();
        implementation.initializeVenmo(getActivity(), new VenmoInitializationCallback() {
            @Override
            public void onInitialized() {
                System.out.println("onInitialized call");
            }
        });
    }

    @PluginMethod
    public void echo(PluginCall call) {
        call.resolve();
    }

    @PluginMethod
    public void startVenmo(PluginCall call) {
        String consumerId = call.getString("consumerId");

        implementation.tokenize(getContext(), consumerId, new VenmoTokenizationCallback() {
            @Override
            public void onTokenized(String result) {
                JSObject ret = new JSObject();
                ret.put("paymentToken", result);
                call.resolve(ret);
            }
        });
    }
}

File: test-plugin-ios-and-java/ios/Sources/PaysafePluginPlugin/PaysafePluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(PaysafePluginPlugin)
public class PaysafePluginPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "PaysafePluginPlugin"
    public let jsName = "PaysafePlugin"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "startVenmo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = PaysafePlugin()
    
    public override func load() {
        implementation.setupSDK()
    }

    @objc func echo(_ call: CAPPluginCall) {
        call.resolve()
    }
    
    @objc func startVenmo(_ call: CAPPluginCall) {
        let consumerId = call.getString("consumerId") ?? ""
        implementation.initializeVenmo { [weak self] in
            self?.implementation.tokenize(consumerId: consumerId) { result in
                call.resolve([
                    "paymentToken": result
                ])
            }
        }
    }
}
